theano/gpuarray/tests/tstgpueye.c

```diff
@@ -34,13 +34,10 @@ int APPLY_SPECIFIC(tstgpueye)(PyArrayObject *n, PyArrayObject *m,
   if (*z == NULL)
     return -1;
 
-  args[0] = (*z)->ga.data;
-  args[1] = &dims[0];
-  args[2] = &dims[1];
   ls = 1;
   gs = 256;
-  /* The k_eye name comes from the kernel declaration above. */
-  err = GpuKernel_call(&k_eye, 1, &ls, &gs, 0, args);
+  /* The eye_call name comes from the kernel declaration above. */
+  err = eye_call(1, &ls, &gs, 0, (*z)->ga.data, dims[0], dims[1]);
```

Comment:
1) Can you give something else that they would like to pass instead of data
for inputs tagged as GpuArray? The shapes/strides will be passed with other
types. The offset must be passed with a different type as well as the
flags. I don't see what else could be passed. A kernel that support only
CUDA and the user would pass the ptr+offset in the same parameter instead
of 2 parameters?

On Thu, Sep 1, 2016 at 1:22 PM, abergeron <notifications@github.com> wrote:

> In theano/gpuarray/tests/tstgpueye.c
> <https://github.com/Theano/Theano/pull/4925#discussion_r77216494>:
>
> >    ls = 1;
> >    gs = 256;
> > -  /* The k_eye name comes from the kernel declaration above. */
> > -  err = GpuKernel_call(&k_eye, 1, &ls, &gs, 0, args);
> > +  /* The eye_call name comes from the kernel declaration above. */
> > +  err = eye_call(1, &ls, &gs, 0, (*z)->ga.data, dims[0], dims[1]);
>
> 1) want to take gpudata pointers to allow people to work with low-level
> gpudata object directly if necessary. Other this will force them to use the
> GpuKernel_call interface for that.
>
> 2) ok.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Theano/Theano/pull/4925/files/3f6653eca5cd7ce8498713a5aa6bc5392262d497#r77216494>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AALC-1rJ1HxRYkxVhgLPgQsu4sdcywSJks5qlwnAgaJpZM4JyKjp>
> .
>